client/api/python/heketi/heketi.py

```diff
@@ -44,13 +52,156 @@ def _set_token_in_header(self, headers, method, uri):
         token = jwt.encode(claims, self.key, algorithm='HS256')
         headers['Authorization'] = 'bearer ' + token
 
+        return headers
+
 
     def hello(self):
         method = 'GET'
         uri = '/hello'
 
         headers={}
-        self._set_token_in_header(headers, method, uri)
+        self._set_token_in_header(method, uri)
         r = requests.get(self.host + uri, headers=headers)
         return r.status_code == requests.codes.ok
 
+
+    def _make_request(self, method, uri, data={}):
+        self.method = method
+        self.uri = uri
+        self.data = data
+
+        headers = self._set_token_in_header(self.method, self.uri)
+
+        ''' Ref: http://docs.python-requests.org/en/master/_modules/requests/api/#request '''
+        r = requests.request(self.method,
+                             self.host + self.uri,
+                             headers=headers,
+                             data=json.dumps(data))
+
+        r.raise_for_status()
+
+
+        if r.status_code == requests.codes.accepted and \
+           r.headers['location']:
+            return self._get_queued_response(r.headers['location'])
+
+        elif r.status_code == requests.codes.SEE_OTHER and \
+             r.headers['location']:
+            return self._get_queued_response(r.headers['location'])
+
+        elif r.status_code == requests.codes.created or \
+             r.status_code == requests.codes.ok:
+            if self.method == 'DELETE':
+                return True
+            else:
+                return r.json()
+
+        elif r.status_code == requests.codes.NO_CONTENT:
+            return True
+
+
+
+    def _get_queued_response(self, queue_uri):
+        self.queue_uri = queue_uri
+        headers = self._set_token_in_header('GET', self.queue_uri)
+        response_ready = False
+
+        while response_ready is False:
+            q = requests.get(self.host + queue_uri,
+                             headers=headers,
+                             allow_redirects=False)
+
+            # Raise an exception when the request fails
+            q.raise_for_status()
+
+
+            if q.status_code == requests.codes.see_other and q.headers['location']:
+                # Temporary request of 303 with header info
+                response_ready = True
+                vol_req = q.headers['location']
+                return self._make_request('GET', vol_req)
+
+            elif q.status_code == requests.codes.NO_CONTENT:
+                # Request successful
+                return True
+                response_ready = True
+
```

Comment:
There should be an ```else``` sleep 1 to 5 seconds.  If not, the server will have to keep answering as fast as the client is requesting.